mozilla-lockbox/lockbox-android
# Explicitly schedule telemetry pings

This PR explicitly schedules telemetry pings to be uploaded in two cases:

* When the Lockbox application moves to the background
* When a change to the `sendUsageData` setting is made

Fixes #230

_(**Required**: this reference (one or many) will be closed upon merge. Ideally it has the acceptance criteria and designs for features or fixes related to the work in this Pull Request.)_

## Testing and Review Notes

_(**Required**: steps to take to confirm this works as expected or other guidance for code, UX, and any other reviewers)_


* Open Lockbox
* Perform some activities that trigger telemetry events (for now it's just startup/foreground/background)
* Trigger an upload by toggling "send usage data" in Settings off/on
* Verify at the Telemetry Portal there are `Lockbox` events

## To Do

- add “WIP” to the PR title if pushing up but not complete nor ready for review
  - optional: consider adding instrumentation (integration/UI) tests
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- ~request the "UX" team perform a design review (if/when applicable)~
- ~check on the [accessibility](https://mozilla-lockbox.github.io/lockbox-android/accessibility/) of any added UI~
